pkg/controller/authentication/authentication_controller.go

```diff
@@ -0,0 +1,268 @@
+func (r *ReconcileAuthentication) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ", "controller", ControllerName)
+
+	ctx := context.Background()
+
+	// Fetch the Authentication spec. If present, we deploy dex in the cluster
+	authentication, err := utils.GetAuthentication(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.OnCRNotFound()
+		}
+		return reconcile.Result{}, err
+	}
+	r.status.OnCRFound()
+	reqLogger.V(2).Info("Loaded config", "config", authentication)
+
+	// Query for the installation object.
+	installationCR, err := installation.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Error(err, "Installation not found")
+			r.status.SetDegraded("Installation not found", err.Error())
+			return reconcile.Result{}, err
+		}
+		log.Error(err, "Error querying installation")
+		r.status.SetDegraded("Error querying installation", err.Error())
+		return reconcile.Result{}, err
+	}
+	if installationCR.Status.Variant != oprv1.TigeraSecureEnterprise {
+		log.Error(err, fmt.Sprintf("Waiting for network to be %s", oprv1.TigeraSecureEnterprise))
+		r.status.SetDegraded(fmt.Sprintf("Waiting for network to be %s", oprv1.TigeraSecureEnterprise), "")
+		return reconcile.Result{}, nil
+	}
+
+	// Make sure Authentication and ManagementClusterConnection are not present at the same time.
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if managementClusterConnection != nil {
+		log.Error(err, "Only one of Authentication and ManagementClusterConnection may be specified")
+		r.status.SetDegraded("Only one of Authentication and ManagementClusterConnection may be specified", err.Error())
+		return reconcile.Result{}, err
+	} else if err != nil {
+		log.Error(err, "Error querying ManagementClusterConnection")
+		r.status.SetDegraded("Error querying ManagementClusterConnection", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Cert used for TLS between voltron and dex when voltron is proxying dex from https://<manager-url>/dex
+	tlsSecret := &corev1.Secret{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: render.DexTLSSecretName, Namespace: render.OperatorNamespace()}, tlsSecret); err != nil {
+		if errors.IsNotFound(err) {
+			// We need to render a new one.
+			tlsSecret = render.CreateDexTLSSecret()
+		} else {
+			log.Error(err, "Failed to read tigera-operator/tigera-dex-tls secret")
+			r.status.SetDegraded("Failed to read tigera-operator/tigera-dex-tls secret", err.Error())
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Dex will be configured with the contents of this secret, such as clientID and clientSecret.
+	idpSecret, err := getIdpSecret(ctx, r.client, authentication)
+	if err != nil {
+		log.Error(err, "Invalid or missing idp secret")
```

Comment:
These are user facing error messages, so we should be spelling out "identity provider" and also providing the name and namespace of the secret that is missing.

The intention is to make it easy for the user to spot what they have misconfigured or what dependencies they have failed to provide.